design/design-doc-template-stacks-experience.md

```diff
@@ -0,0 +1,473 @@
+
+```
+kubectl crossplane stack install myorg/mystack mystack
+```
+
+This installs the stack from the `myorg/mystack` image, using the name `mystack`.
+
+#### Create
+
+Instantiation is also very similar to any stack:
+
+```
+cat > my-instance.yaml <<EOF
+apiVersion: thing.samples.stacks.crossplane.io/v1alpha1
+kind: ThingInstance
+metadata:
+  name: thinginstance-sample
+spec:
+  myfield: myvalue
+EOF
+
+kubectl apply -f my-instance.yaml
+```
+
+The difference is that underneath, the ThingInstance will be used as the
+input for rendering a template.
+
+#### Delete
+
+When the ThingInstance is deleted, the corresponding resources will also
+be deleted. This is the same behavior as in any stack.
+
+```
+cat > my-instance.yaml <<EOF
+apiVersion: thing.samples.stacks.crossplane.io/v1alpha1
+kind: ThingInstance
+metadata:
+  name: thinginstance-sample
+spec:
+  myfield: myvalue
+EOF
+
+kubectl delete -f my-instance.yaml
+```
+
+The instance is also deleted if the stack is uninstalled.
+
+#### Uninstall
+
+Uninstall looks the same as any other stack:
+
+```
+kubectl crossplane stack uninstall mystack
+```
+
+#### Upgrade and update
+
+Upgrading a stack version, and updating an instance of an object managed
+by the stack, are out of scope of this document. However, one could
+imagine that the process of upgrading a stack version would change a
+version number on object instances which go with the stack, and that the
+version number change would underneath cause the template to be rendered
+again and applied to the cluster.
+
+### The stack.yaml
+
+Configuration for the template stack will be in a `stack.yaml` file at
+the build root of the stack in the repository. For most cases, this will
+mean the `stack.yaml` lives at the root of the repository.
+
+#### Specifying how to process render requests
+
+A Stack author must create a `stack.yaml` in order to configure how
+templates are rendered in response to a render request. Here's an
+example, with comments explaining the directives:
+
+```
+cat > stack.yaml <<EOF
+# This field configures which templates are rendered in response
+# to a given object type. An instance of the object type is
+# considered to be a "render request".
+behaviors:
+  crds:
+    # This is a particular CRD which is being configured. When the
+    # controller sees an object of this type, it will do something.
+    wordpressinstance.wordpress.samples.stacks.crossplane.io/v1alpha1:
+      # These are the templates which should be rendered when an object
+      # of the type above is seen.
+      #
+      # Defaults for the variables can be defined in the default values
+      # for the CRD fields, using the standard Kubernetes mechanism for
+      # specifying CRD field default values.
+      resources:
+        # These are individual files which will be rendered.
+        - templates/kubernetescluster.yaml
+        - templates/mysqlinstance.yaml
+        - tempaltes/kubernetesapplication.yaml
```

Comment:
`templates` typo

Are you concerned that by not using a name mapping to template files that we will lose the ability to map template derived resource values within templates `{{ .templateName.resourceValue }}`?

I realize there are alternatives:
* `{{ template("template/path").resourceValue }}`
* `{{ resource("kind", "name").resourceValue }}` 
   _note: this would not create a GET API request, this would be rendered from an pre-rendering fetch of all the template's known resources_
* others?

I haven't personally pinned one down and I appreciate your investigation into this part of the design.